ecosystem/sep-0010.md

```diff
@@ -271,6 +281,29 @@ A server may choose to issue JWTs for less than all thresholds and based on any
 
 A server that needs to support validating accounts that do not exist can require a signature of the master key of the account address for accounts that do not exist.
 
+### Verifying Client Application Identity
+
+A server may want to attribute each HTTP request to a specific client application. For example, a server and client may agree to change fees for the users of that client. SEP-10 offers an optional variation to the flow outlined in the [Abstract](#abstract) to enable this use case. 
+
+In order to use this optional feature, the organization that owns the client application must host a [SEP-1 stellar.toml](sep-0001.md) file containing a `SIGNING_KEY` attribute. This allows the SEP-10 server to verify that the challenge returned by the client is signed with the `SIGNING_KEY`. This proves that the client is associated with the client domain. The SEP-10 server can now attribute requests made with the resulting JWT to the client domain that signed the challenge.
+
+While SEP-10 servers may chose to verify all client applications who request attribution, it is highly recommended to only verify a specific set of client applications and ignore attribution requests from other clients. **Making requests to unknown domains may slow down or halt the SEP-10 server's ability to respond to other requests.**
+
+Outlined below is the flow between client applications and SEP-10 servers when attribution is requested:
+
+* The client makes the initial `GET <WEB_AUTH_ENDPOINT>` request with an added `?client_domain=example.com` parameter and value
+* If the SEP-10 server supports attribution requests for the specified `client_domain`, it fetches the `SIGNING_KEY` from the `client_domain`'s [SEP-1 stellar.toml](sep-0001.md) and generates the challenge transaction with an additional Manage Data operation as described in the [Response](#response) section, referred to here as the `client_domain` Manage Data operation
+  * If the SEP-10 server is unable to fetch the `SIGNING_KEY`, return a `424 Failed Dependency` HTTP status code
```

Comment:
I think it should fail, because there is a difference between the server choosing not to support attribution for the provided client domain and being unable to support attribution for the provided client domain. One is the server's prerogative, and the other is a failure of the client to make attribution possible.

I'm not tied to `424`, but it did seem to perfectly fit this situation. Another choice may be `422`, but that also seems to be defined within the WebDAV context. However, 422 is definitely used outside the WebDAV context in the wild. 

`400 Bad Request` is not a good fit here because the request can be well formed and the server would still return an error if the `SIGNING_KEY` couldn't be obtained.

The best option remaining may be `409 Conflict`

Reply:
If this is going to error, I think a `400` is fine. [RFC7231](https://tools.ietf.org/html/rfc7231#section-6.5.1) superseded [RFC2616](https://tools.ietf.org/html/rfc2616#section-10.4.1) and broadened the scope of the status code to apply to any request where the server won't process the request because it perceives the client made a mistake. There could be multiple reasons for why a `SIGNING_KEY` is not fetchable. Maybe the client specified the wrong domain, maybe the domain is currently down, but the domain is provided by the client organization so I think we can say that it is the clients fault if the domain they wish to verify is not setup appropriately.

Reply:
Ok, I'm alright with that 👍